app/db/migrations/20210506170155-update-user-profile-url.js

```diff
@@ -0,0 +1,32 @@
+'use strict'
+
+module.exports = {
+  up: async (queryInterface, Sequelize) => {
+    // Increases the length of a social profile image URL beyond the default
+    // value of 255 characters. Although there's no standard maximum length
+    // of a URL, a common rule of thumb is that URLs should not be longer than
+    // 2083 as this is the maximum length supported by IE. Despite this, it
+    // should be reasonable to expect that the URLs in this particular use case
+    // would not be longer than 1024 characters. See the issue at
+    // https://github.com/streetmix/streetmix/issues/2335 for more information.
+    await queryInterface.changeColumn('Users', 'profile_image_url', {
+      type: Sequelize.STRING(1024)
+    })
+  },
+
+  down: async (queryInterface, Sequelize) => {
+    // If rolling back, this should set the column back to the original STRING
+    // definition, with a maximum value of 255 characters.
+    //
+    // According to PostgreSQL documentation:
+    // "...an over-length value will be truncated to n characters without
+    // raising an error. (This too is required by the SQL standard.)"
+    // https://www.postgresql.org/docs/current/datatype-character.html
```

Comment:
Trying to actually roll back the migration with a 265-character profile image URL netted me this error from Postgres:

```
➜  streetmix git:(louh/issue-2335) ✗ npx sequelize-cli db:migrate:undo

Sequelize CLI [Node: 14.16.1, CLI: 6.2.0, ORM: 6.6.2]

Loaded configuration file "app/db/config/config.js".
Using environment "development".
== 20210506170155-update-user-profile-url: reverting =======

ERROR: value too long for type character varying(255)
```

I looked at that documentation you included (and am so happy you did, else I'd have been lost a bit longer), and I think that snippet is unfortunately only about casting to those types? This seems to be the more worryingly relevant bit:

> An attempt to store a longer string into a column of these types will result in an error, unless the excess characters are all spaces, in which case the string will be truncated to the maximum length. (This somewhat bizarre exception is required by the SQL standard.)

It's frustrating (but understandable) that we can't opt into truncation. In the interest of clean, reversible migrations, I think we'll have to perform a pre-flight that manually truncates any >255-character values, and then changes the type.

Reply:
Thank you for testing this, and finding a solution!